### PR TITLE
Improve the help message of 'tools/build.py'

### PR DESCRIPTION
**Pull request by LaszloLango**
Added groups with descriptions and sorted the agruments.

IoT.js-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by daeyeon** [comment]
Please fix the conflicts.

File: tools/build.py [hunk 1]
```diff
@@ -71,130 +71,130 @@ def init_options():
...
+        'the ECMA-262 standard.')
+    jerry_group.add_argument('--jerry-cmake-param',
         action='append', default=[],
         help='Specify additional cmake parameters for JerryScript '
         '(can be used multiple times')
-    parser.add_argument('--jerry-compile-flag',
+    jerry_group.add_argument('--jerry-compile-flag',
         action='append', default=[],
         help='Specify additional compile flags for JerryScript '
              '(can be used multiple times')
```
Comment (daeyeon) [inline tools/build.py]: ditto

File: tools/build.py [hunk 2]
```diff
@@ -71,130 +71,130 @@ def init_options():
...
-    parser.add_argument('--jerry-cmake-param',
+
+    jerry_group = parser.add_argument_group('Arguments of JerryScript',
+        'The following arguments are related to the JavaScript engine under '
+        'the framework. For example they can change the enabled features of '
+        'the ECMA-262 standard.')
+    jerry_group.add_argument('--jerry-cmake-param',
         action='append', default=[],
         help='Specify additional cmake parameters for JerryScript '
         '(can be used multiple times')
```
Comment (daeyeon) [inline tools/build.py]: ')' is missing at the end. Please fix this also.

File: tools/build.py [hunk 3]
```diff
@@ -71,130 +71,130 @@ def init_options():
...
+    iotjs_group.add_argument('--target-board',
+        choices=[None, 'artik10', 'stm32f4dis', 'rpi2', 'rpi3', 'artik05x'],
+        default=None, help='Specify the target board (if needed): '
+             '%(choices)s (default: %(default)s)')
+    iotjs_group.add_argument('--target-os',
+        choices=['linux', 'darwin', 'osx', 'nuttx', 'tizen', 'tizenrt',
+                 'openwrt'],
+        default=platform.os(),
+        help='Specify the target os: %(choices)s (default: %(default)s)')
+    iotjs_group.add_argument('--test-driver',
```
Comment (daeyeon) [inline tools/build.py]: This looks dead option.

**Comment by LaszloLango** [comment]
@daeyeon thanks for the review. I've updated the PR.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**